File: py_to_c.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <json-c/json.h>

#define JSON_FILE "data.json"

void read_json_file(const char *filename) {
    struct json_object *parsed_json;
    struct json_object *donnees_recues;
    FILE *fp = fopen(filename, "r");
    char buffer[1024];

    if (fp == NULL) {
        perror("Erreur lors de l'ouverture du fichier JSON");
        return;
    }

    fread(buffer, 1024, 1, fp);
    fclose(fp);

    parsed_json = json_tokener_parse(buffer);

    // Extraction des données du fichier JSON
    json_object_object_get_ex(parsed_json, "donnees_recues", &donnees_recues);

    // Utiliser les données extraites du JSON comme nécessaire
    printf("Données reçues : %s\n", json_object_to_json_string_ext(donnees_recues, JSON_C_TO_STRING_PRETTY));
}

int main() {
    while (1) {
        read_json_file(JSON_FILE);

        // Pause pour espacer les itérations (attendre que le fichier soit mis à jour)
        sleep(1);
    }

    return 0;
}

File: py_to_c.py
import json
import random
import time

JSON_FILE = 'data.json'

def write_json_file(filename, data):
    with open(filename, 'w') as f:
        json.dump(data, f, indent=4)

def generate_random_data(num_entries):
    return [random.randint(0, 99999) for _ in range(num_entries)]

def main():
    while True:
        # Génération de données aléatoires
        random_data = generate_random_data(5)
        
        # Création de l'objet JSON
        data = {
            "donnees_recues": random_data
        }

        # Écriture dans le fichier JSON
        write_json_file(JSON_FILE, data)
        print(f"Données aléatoires écrites sur le fichier JSON : {json.dumps(random_data, indent=4)}")
        
        # Pause pour espacer les itérations
        time.sleep(1)

if __name__ == "__main__":
    main()
